decentraland/unity-explorer
# chore: stop raycasts outside of scenes

## What does this PR change?

1. Stops raycasts on outside scenes (`PBRaycastResults` and `InteractionHighlights`)
2. Refactor to `UpdateSceneSystem`. It was marking a scene as current on every frame, generating garbage, unneded processing and possible inconsistencies

Before
![image](https://github.com/decentraland/unity-explorer/assets/1999557/c5401f5f-e812-4835-9046-7a8922732aff)

After
![image](https://github.com/decentraland/unity-explorer/assets/1999557/7c8dd166-5594-4216-8633-727b74480721)


## How to test the changes?

1. Go to Goerli
2. There is a scene in 82,-5 (the mirrors). Look at the mirror from inside the scene in first person mode and mouse locked. A tooltip should appear
3. Look at it form outside the scene. The tooltip should not appear
4. Look at it, and try to look at it while you leave the scene. The tooltip should dissapear

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
